handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/ServiceCombServerStats.java

```diff
@@ -84,20 +84,35 @@ private void init(){
   }
 
   public static boolean isolatedServerCanTry() {
-    return globalAllowIsolatedServerTryingFlag.get();
+    TryingIsolatedServerMarker marker = globalAllowIsolatedServerTryingFlag.get();
+    if (marker == null) {
+      return true;
+    }
+    return marker.isOutdated();
   }
 
   /**
    * Applying for a trying chance for the isolated server. There is only 1 trying chance globally concurrently.
    *
    * @return true if the chance is applied successfully, otherwise false
    */
-  public static boolean applyForTryingChance() {
-    return isolatedServerCanTry() && globalAllowIsolatedServerTryingFlag.compareAndSet(true, false);
+  public static boolean applyForTryingChance(Invocation invocation) {
```

Comment:
This method is not thread safe.

Reply:
The thread safty is ensured by the CAS operation `AtomicReference#compareAndSet`